File: datasets/rb/HumanEval_28_concatenate.rb
# Concatenate list of strings into a single string
# >>> concatenate([])
# ''
# >>> concatenate(['a', 'b', 'c'])
# 'abc'
def concatenate(strings)
  strings.join
end

require 'test/unit'
class TestHumanEval < Test::Unit::TestCase
  def test_concatenate
    candidate = method(:concatenate)
    assert_equal('', candidate.call([]))
    assert_equal('xyz', candidate.call(['x', 'y', 'z']))
    assert_equal('xyzwk', candidate.call(['x', 'y', 'z', 'w', 'k']))
  end
end

File: datasets/lua-keep-incoder_6b_temp_0.2-3/HumanEval_28_concatenate.lua
-- Concatenate list of strings into a single string
-- >>> concatenate([])
-- ''
-- >>> concatenate(['a', 'b', 'c'])
-- 'abc'
local function concatenate(strings)
  local result = ""
  for _, string in ipairs(strings) do
    result = result .. string
  end
  return result
end


lu = require('luaunit')

function test_humaneval()
local candidate = concatenate
    lu.assertEquals(candidate({}), '')
    lu.assertEquals(candidate({'x', 'y', 'z'}), 'xyz')
    lu.assertEquals(candidate({'x', 'y', 'z', 'w', 'k'}), 'xyzwk')
end

os.exit(lu.LuaUnit.run())